internal/validators/secretref/secret_ref.go

```diff
@@ -84,3 +97,101 @@ func checkForMissingFields(ref telemetryv1alpha1.SecretKeyRef) error {
 
 	return nil
 }
+
+func getSecretRefsTracePipeline(tp *telemetryv1alpha1.TracePipeline) []telemetryv1alpha1.SecretKeyRef {
+	return getRefsInOTLPOutput(tp.Spec.Output.OTLP)
+}
+
+func getSecretRefsMetricPipeline(mp *telemetryv1alpha1.MetricPipeline) []telemetryv1alpha1.SecretKeyRef {
+	return getRefsInOTLPOutput(mp.Spec.Output.OTLP)
+}
+
+func getSecretRefsLogPipeline(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
+	var refs []telemetryv1alpha1.SecretKeyRef
+
+	for _, v := range lp.Spec.Variables {
+		if v.ValueFrom.SecretKeyRef != nil {
+			refs = append(refs, *v.ValueFrom.SecretKeyRef)
+		}
+	}
+
+	refs = append(refs, getEnvSecretRefs(lp)...)
+	refs = append(refs, getTLSSecretRefs(lp)...)
+
+	return refs
+}
+
+// getEnvSecretRefs returns the secret references of a LogPipeline that should be stored in the env secret
+func getEnvSecretRefs(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
+	var refs []telemetryv1alpha1.SecretKeyRef
+
+	output := lp.Spec.Output
+	if output.HTTP != nil {
+		refs = appendIfSecretRef(refs, output.HTTP.Host)
+		refs = appendIfSecretRef(refs, output.HTTP.User)
+		refs = appendIfSecretRef(refs, output.HTTP.Password)
+	}
+
+	return refs
+}
+
+func getTLSSecretRefs(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
```

Comment:
You already have a function `getSecretRefsInOTLPOutput`. Maybe just merge those 2 functions into `getSecretRefsInHTTPOutput` for the sake of consistency?